doc/source/development/testing.rst

```diff
@@ -121,3 +121,172 @@ follows:
    reference the suppression files and other common arguments in a
    ``~/.valgrindrc`` file.
 
+
+Recommendations on how to write new tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Python-based tests should be preferred when possible, as productivity is higher
+in Python and there is no associated compilation time (compilation time affects
+feedback received from continuous integration).
+
+C/C++-based test should be reserved for C++-specific aspects that cannot be tested
+with the SWIG Python bindings, which use the C interface. For example testing
+of C++ operators (copy/move constructors/assignment operators, iterator interfaces,
+etc.) or C/C++ functionality not mapped to SWIG (e.g. CPL utility functions/classes)
+
+Python tests
+------------
+
+Python tests use the `pytest <https://docs.pytest.org/en/latest/>`__
+framework since :ref:`rfc-72`.
+
+Test cases should be written in a way where they are independent from other
+ones, so they can potentially be run in a isolated way or in parallel of other
+test cases. In particular temporary files should be created with a name that
+cannot conflict with other tests.
```

Comment:
Preferably use the [`tmp_path` fixture](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture)

Reply:
```suggestion
test cases. In particular temporary files should be created with a name that
cannot conflict with other tests: preferably use pytest's [`tmp_path` fixture](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture).
```